packages/plugin-ext-vscode/src/node/plugin-vscode-decompress.ts

```diff
@@ -0,0 +1,36 @@
+// *****************************************************************************
+// Copyright (C) 2023 EclipseSource and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import * as decompress from 'decompress';
+import * as path from 'path';
+import * as fs from '@theia/core/shared/fs-extra';
+
+export async function decompressExtension(sourcePath: string, destPath: string): Promise<boolean> {
+    try {
+        await decompress(sourcePath, destPath);
+        if (sourcePath.endsWith('.tgz')) {
+            const extensionPath = path.join(destPath, 'package');
```

Comment:
A comment to explain this path mangling would be nice

Reply:
I just moved this part of code over there from https://github.com/eclipse-theia/theia/blob/2d30b295c1158f401d63591eb67b46e93cd6f318/packages/plugin-ext-vscode/src/node/plugin-vscode-file-handler.ts#L76

I had no idea for which exact use case we needed this, but I did not want to create regressions, so I kept it. Comes from [this commit](https://github.com/eclipse-theia/theia/commit/ccaa5d84c5718ce66e887b7bed620366fc859c11) originally. It seems that it unpacks an additional `vscode_node_modules.zip` file in case the source file was a .tgz. I can add a comment that says so, but I do not know the exact use case.